### DISCUSSION TITLE
invalid OpenAPI specs file

### DISCUSSION THREAD

**Discussion by samar-elsayed** [apache/camel-karavan#1220] [Q&A] [answered]
I am trying to create OpenAPI spec file for the rest endpoint created in the project via Karavan VSCode (latest version)
and here for the Inquiry Path I have created 2 params 

![image](https://github.com/apache/camel-karavan/assets/22873802/1e11c25e-1703-4b66-a416-829f580a3870)

![image](https://github.com/apache/camel-karavan/assets/22873802/8ceeecff-1c56-4d1f-9397-186744415533)

and both are required but as shown below when I check the doc path I find only one param of them is mentioned and it doesn't look correct (there is example value for clientdate that doesn't show up in the file )

![image](https://github.com/apache/camel-karavan/assets/22873802/ba6491ea-29a9-49c1-8dda-66756faba36f)

the OpenAPI file : 
![image](https://github.com/apache/camel-karavan/assets/22873802/bcec0168-daf1-46a0-90d9-71f67ea6b34d)

when I check that file inside swagger editor:

![image](https://github.com/apache/camel-karavan/assets/22873802/1f3b2916-717e-4721-bb8c-48f2151fe998)

**Comment by mgubaidullin** [comment, marked as answer]
Karavan and Camel are not tools to create OpenAPI specs.
You can create and run a REST service but it will not provide full OpenAPI spec.

**Reply by davsclaus** [reply]
Camel 4.6 onwards will come with contract first rest-dsl that allows to use a openapi spec file as-is. Then you can do all the openapi stuff you need.
